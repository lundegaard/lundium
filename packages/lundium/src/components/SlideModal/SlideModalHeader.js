import React from 'react';
import PropTypes from 'prop-types';
import { cx } from 'ramda-extension';

import Grid from '../Grid';
import Text from '../Text';

const SlideModalHeader = ({
	children,
	className,
	title,
	subtitle,
	details,
	...otherProps
}) => (
	<Grid
		justifyContent="between"
		alignItems="end"
		flexWrap
		className={cx('slide-modal__header', className)}
		{...otherProps}
	>
		<Text className="slide-modal__title">{title}</Text>
		{subtitle && <Text className="slide-modal__sub-title">{subtitle}</Text>}
		{details && (
			<Grid
				flex
				flexWrap
				flexDirection={['column', 'row']}
				className="slide-modal__details"
			>
				{details}
			</Grid>
		)}
		{children}
	</Grid>
);

SlideModalHeader.propTypes = {
	/** Children to be rendered in the main container. */
	children: PropTypes.node,
	/** Class name for root. */
	className: PropTypes.string,
	details: PropTypes.node,
	subtitle: PropTypes.node,
	title: PropTypes.node.isRequired,
};

export default SlideModalHeader;
