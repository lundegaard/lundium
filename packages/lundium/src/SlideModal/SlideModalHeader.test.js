import React from 'react';
import { shallow } from 'enzyme';

import SlideModalHeader from './SlideModalHeader';

describe('<SlideModalHeader />', () => {
	const requiredProps = { title: 'Modal heading' };

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<SlideModalHeader {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
