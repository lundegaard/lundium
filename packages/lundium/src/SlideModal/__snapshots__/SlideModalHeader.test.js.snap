// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SlideModalHeader /> matches snapshot when shallowly rendered 1`] = `
<forwardRef(Grid)
  alignItems="end"
  className="slide-modal__header"
  flexWrap={true}
  justifyContent="between"
>
  <forwardRef(Text)
    className="slide-modal__title"
  >
    Modal heading
  </forwardRef(Text)>
</forwardRef(Grid)>
`;
